kbs/src/api_server.rs

```diff
@@ -0,0 +1,317 @@
+
+                let claims = core.token_verifier.verify(token).await?;
+
+                let claim_str = serde_json::to_string(&claims)?;
+                if !core
+                    .policy_engine
+                    .evaluate(resource_desc, &claim_str)
+                    .await?
+                {
+                    return Err(Error::PolicyDeny);
+                };
+
+                let resource_description = ResourceDesc::try_from(resource_desc)?;
+                let resource = core
+                    .resource_storage
+                    .get_secret_resource(resource_description)
+                    .await?;
+
+                let public_key = core.token_verifier.extract_tee_public_key(claims)?;
+                let jwe = jwe(public_key, resource).map_err(|e| Error::JweError { source: e })?;
+
+                let res = serde_json::to_string(&jwe)?;
+
+                Ok(HttpResponse::Ok()
+                    .content_type("application/json")
+                    .body(res))
+            } else if request.method() == Method::POST {
+                let resource_desc =
+                    sub_path
+                        .strip_prefix('/')
+                        .ok_or(Error::IllegalAccessedPath {
+                            path: end_point.clone(),
+                        })?;
+                let resource_description = ResourceDesc::try_from(resource_desc)?;
+                core.admin_auth.validate_auth(&request)?;
+                core.resource_storage
+                    .set_secret_resource(resource_description, &body)
+                    .await?;
+
+                Ok(HttpResponse::Ok().content_type("application/json").body(""))
+            } else {
+                Ok(HttpResponse::NotImplemented()
+                    .content_type("application/json")
+                    .body(""))
+            }
+        }
+        plugin_name => {
+            // Plugin calls needs to be authorized by the Token and policy
+            let token = core
+                .get_attestation_token(&request)
+                .await
+                .map_err(|_| Error::TokenNotFound)?;
+
+            let claims = core.token_verifier.verify(token).await?;
+
+            let claim_str = serde_json::to_string(&claims)?;
+
+            // TODO: add policy filter support for other plugins
+            if !core.policy_engine.evaluate(&end_point, &claim_str).await? {
+                return Err(Error::PolicyDeny);
+            }
+
+            let plugin = core
+                .plugin_manager
+                .get(plugin_name)
+                .ok_or(Error::PluginNotFound {
+                    plugin_name: plugin_name.to_string(),
+                })?;
+            let body = body.to_vec();
+            let response = plugin
+                .handle(body, query.into(), sub_path.into(), request.method())
+                .await?;
+            Ok(response)
+        }
+    }
+}
+
+/// Admin APIs.
+pub(crate) async fn admin(
```

Comment:
This is a bit confusing since we have admin APIs in the client section above as well

Reply:
I want to move all admin APIs into this section. Let converge onto https://github.com/confidential-containers/trustee/pull/514#discussion_r1779304844